acceptance/lib/puppet_x/acceptance/pki.rb

```diff
@@ -0,0 +1,222 @@
+
+        cert
+      end
+
+      def self.create_csr(key, name)
+        csr = OpenSSL::X509::Request.new
+
+        csr.public_key = key.public_key
+        csr.subject = OpenSSL::X509::Name.parse(name)
+        csr.version = 2
+        csr.sign(key, DEFAULT_SIGNING_DIGEST)
+
+        csr
+      end
+
+      def self.sign(ca_key, ca_cert, csr, extensions = NODE_EXTENSIONS)
+        cert = OpenSSL::X509::Certificate.new
+
+        cert.public_key = csr.public_key
+        cert.subject = csr.subject
+        cert.issuer = ca_cert.subject
+        cert.version = 2
+        cert.serial = rand(2**128)
+
+        not_before = just_now
+        cert.not_before = not_before
+        cert.not_after = not_before + FIVE_YEARS
+
+        ext_factory = extension_factory_for(ca_cert, cert)
+        extensions.each do |ext|
+          extension = ext_factory.create_extension(*ext)
+          cert.add_extension(extension)
+        end
+
+        cert.sign(ca_key, DEFAULT_SIGNING_DIGEST)
+
+        cert
+      end
+
+      def self.create_crl_for(ca_cert, ca_key)
+        crl = OpenSSL::X509::CRL.new
+        crl.version = 1
+        crl.issuer = ca_cert.subject
+
+        ef = extension_factory_for(ca_cert)
+        crl.add_extension(
+          ef.create_extension(["authorityKeyIdentifier", "keyid:always", false]))
+        crl.add_extension(
+          OpenSSL::X509::Extension.new("crlNumber", OpenSSL::ASN1::Integer(0)))
+
+        not_before = just_now
+        crl.last_update = not_before
+        crl.next_update = not_before + FIVE_YEARS
+        crl.sign(ca_key, DEFAULT_SIGNING_DIGEST)
+
+        crl
+      end
+
+      def self.revoke(serial, crl, ca_key)
+        revoked = OpenSSL::X509::Revoked.new
+        revoked.serial = serial
+        revoked.time = Time.now
+        revoked.add_extension(
+          OpenSSL::X509::Extension.new("CRLReason",
+                                       OpenSSL::ASN1::Enumerated(DEFAULT_REVOCATION_REASON)))
+
+        crl.add_revoked(revoked)
+        extensions = crl.extensions.group_by{|e| e.oid == 'crlNumber' }
+        crl_number = extensions[true].first
+        unchanged_exts = extensions[false]
+
+        next_crl_number = crl_number.value.to_i + 1
+        new_crl_number_ext = OpenSSL::X509::Extension.new("crlNumber",
+                                                          OpenSSL::ASN1::Integer(next_crl_number))
+
+        crl.extensions = unchanged_exts + [new_crl_number_ext]
+        crl.sign(ca_key, DEFAULT_SIGNING_DIGEST)
+
+        crl
+      end
+
+      # Creates a self-signed root ca, then signs two node certs, revoking one of them.
+      # Creates an intermediate CA and one node cert off of it.
+      # Creates a second intermediate CA and one node cert off if it.
+      # Creates a leaf CA off of the intermediate CA, then signs two node certs revoking one of them.
+      # Revokes an intermediate CA.
+      # Returns the ca bundle, crl chain, and all the node certs
```

Comment:
Whoops, this should go away since I changed the certs that are returned from this method.

Reply:
Looks like this still needs to be removed?